pkgs/native_assets_cli/lib/src/model/os.dart

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of '../api/os.dart';
+
+final class OSImpl implements OS {
```

Comment:
If this would be a sealed class, and all the `static const OSImpl` would be subclasses, the `static const _dylibExtension` and others could be exhaustive switches instead of maps, reducing the number of `!` in the code :)